lib/ui/text.dart

```diff
@@ -833,28 +833,66 @@ class TextBox {
   String toString() => 'TextBox.fromLTRBD(${left.toStringAsFixed(1)}, ${top.toStringAsFixed(1)}, ${right.toStringAsFixed(1)}, ${bottom.toStringAsFixed(1)}, $direction)';
 }
 
-/// Whether a [TextPosition] is visually upstream or downstream of its offset.
+/// Disambiguates cases where a string offset could match two locations in the
+/// rendered string.
```

Comment:
nit: Generally prefer to document types using a noun-phrase as the summary sentence.

https://www.dartlang.org/guides/language/effective-dart/documentation#prefer-starting-library-or-type-comments-with-noun-phrases